spec/lita-reviewme/github_spec.rb

```diff
@@ -0,0 +1,95 @@
+require "spec_helper"
+
+describe Lita::Reviewme::Github do
+  let(:repo) { "gh_user/repo" }
+  let(:id) { "123" }
+  let(:user) { OpenStruct.new(login: 'iamvery') }
+  let(:title) { 'PR Title' }
+  let(:pull_request) { OpenStruct.new(user: user, title: title) }
+  let(:github_comment_template) { ":eyes: %{reviewer}" }
+  let(:github_comment) { true }
+  let(:github_request_review) { false }
+
+  let(:config) do
+    OpenStruct.new(
+      github_access_token: 'access-token',
+      github_comment_template: github_comment_template,
+      github_comment: github_comment,
+      github_request_review: github_request_review
+    )
+  end
+
+  before do
+    allow_any_instance_of(Octokit::Client).to receive(:pull_request)
+      .and_return(pull_request)
+  end
+
+  subject { described_class.new(config, repo, id) }
+
+  describe "ownership" do
+    it "from the pull request" do
+      expect(subject.owner).to eq('iamvery')
+    end
+
+    describe "unexpected error" do
+      class User; def login; raise Octokit::Error; end; end
+
+      let(:user) { User.new }
+
+      it "raises an exception" do
+        expect { subject.owner }.to raise_error(Lita::Reviewme::Github::UnknownOwner)
+      end
+    end
+  end
+
+  describe "assignment" do
+    describe "requested not to comment or request reviewer" do
+      let(:github_comment) { false }
+
+      it "raises an exception" do
+        expect { subject.assign('iamvery') }.to raise_error(Lita::Reviewme::Github::CannotPostComment)
+      end
+    end
+
+    describe "via review request" do
+      let(:github_comment) { false }
+      let(:github_request_review) { true }
+
+      it "uses the request review api" do
+        expect_any_instance_of(Octokit::Client).to receive(:request_pull_request_review)
+          .with(repo, id, ['iamvery'], {:accept=>"application/vnd.github.black-cat-preview"})
```

Comment:
Not sure how I feel about this much implementation detail leaking out in the test.